File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }while (dollars < 0);
    
    
    int cents = round(dollars * 100);
    int coins = 0;
    
    int denominations[] = {25, 10, 5, 1};
    int size = 4;
    for (int i = 0; i < size; i++)
    {
        coins += cents / denominations[i];
        cents %= denominations[i];
    }
    printf("%i\n", coins);    
}

File: pset6/cash/cash.py
from cs50 import get_float

#request input for change owed
while True:
    x = get_float("Change owed: ")
    if x >= 0:
        break

#convert to cents
x = round(x * 100)

count = 0

#keep track of coins used
while x > 0:
    if x >= 25:
        x -= 25
        count += 1
    elif x >= 10:
        x -= 10
        count += 1
    elif x >= 5:
        x -= 5
        count += 1
    elif x >= 1:
        x -= 1
        count += 1

#print change
print(count)